### PR TITLE
Makes action buttons call the correct procs

### LINKED ISSUE #30916: Fire takes priority in strange circumstances.
I had the opportunity to be on fire and snared with a bola, but I was wearing fireproof  clothes. Clicking on the snared action button caused me to stop drop and roll instead of taking the cuffs off.

### PR DESCRIPTION
**Pull request by ShizCalev**
Fixes #30916

They were previously just calling resist, which lead to some priority issues if you intended to stay on fire, but just wanted to unhandcuff yourself.

Now clicking on an action button will now try to perform that specific action instead (if you're able to perform it.)

### CODE REVIEW

**Comment by MimicFaux** [comment]
Speedmerge pls

**Comment by Cyberboss** [comment]
No

File: code/_onclick/hud/alert.dm [hunk 1]
```diff
@@ -646,7 +650,23 @@ so as to remain in compliance with the most up-to-date laws."
...
+	if(isliving(usr))
+		var/mob/living/L = usr
+		if(!can_do_alert_action(L))
+			return
+		L.changeNext_move(CLICK_CD_RESIST)
+		if(L.last_special <= world.time)
+			return L.resist_buckle()
+
+/obj/screen/alert/proc/can_do_alert_action(mob/living/L)
+	return (!((L.next_move > world.time) || L.incapacitated(ignore_restraints = TRUE)))
```
Comment (Cyberboss) [inline code/_onclick/hud/alert.dm]: Just make this a global or mob proc

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -548,7 +548,7 @@
 	set name = "Resist"
 	set category = "IC"
 
-	if(!isliving(src) || next_move > world.time || incapacitated(ignore_restraints = 1))
+	if(!isliving(src) || next_move > world.time || incapacitated(ignore_restraints = TRUE)) //make sure /obj/screen/alert/proc/can_do_alert_action() is updated if changed.
```
Comment (Cyberboss) [inline code/modules/mob/living/living.dm]: So you can call it here

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**